core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -1024,8 +1022,8 @@ func (o *evmTxStore) FindTxesPendingCallback(ctx context.Context, blockNum int64
 	INNER JOIN evm.tx_attempts ON evm.txes.id = evm.tx_attempts.eth_tx_id
 	INNER JOIN evm.receipts ON evm.tx_attempts.hash = evm.receipts.tx_hash
 	WHERE evm.txes.pipeline_task_run_id IS NOT NULL AND evm.txes.signal_callback = TRUE AND evm.txes.callback_completed = FALSE
-	AND evm.receipts.block_number <= ($1 - evm.txes.min_confirmations) AND evm.txes.evm_chain_id = $2
-	`, blockNum, chainID.String())
+	AND evm.receipts.block_number <= $1 AND evm.txes.evm_chain_id = $2
```

Comment:
double check on this, is this what we want when finality tag turned on ? 

If we want to do backward compatibility support, for chain without finality enabled, are we okay with this returning the finalizedHeahNum with formula from calculateLatestFinalized() ? 

Might want to have this discussion on the other [jira](https://smartcontract-it.atlassian.net/browse/BCI-3574) ?